src/backoff.js

```diff
@@ -0,0 +1,387 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/*!
+ * Injected.
+ *
+ * @see Firestore
+ */
+let Firestore;
+
+/*
+ * @module firestore/backoff
+ * @private
+ *
+ * Contains backoff logic to facilitate RPC error handling. This class derives
+ * its implementation from the Firestore Mobile Web Client.
+ *
+ * @see https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/remote/backoff.ts
+ */
+
+/*!
+ * Initial backoff time in milliseconds after an error.
+ * Set to 1s according to https://cloud.google.com/apis/design/errors.
+ */
+const BACKOFF_INITIAL_DELAY_MS = 1000;
+
+/*!
+ * Maximum backoff time in milliseconds.
+ */
+const BACKOFF_MAX_DELAY_MS = 60 * 1000;
+
+/*!
+ * The factor to increase the backup by after each failed attempt.
+ */
+const BACKOFF_FACTOR = 1.5;
+
+/*!
+ * The jitter to distribute the backoff attempts by.
```

Comment:
I kept it as is for now but updated some of the comments. I agree it would be better to keep it in sync, but the code can still be easily matched to the original implementation.